src/particle.cpp

```diff
@@ -404,7 +404,8 @@ void Particle::cross_surface()
     write_message(1, "    Crossing surface {}", surf->id_);
   }
 
-  if (surf->surf_source_ && simulation::current_batch == settings::n_batches) {
+  if (surf->surf_source_ && simulation::current_batch > settings::n_inactive &&
+      !simulation::surf_source_bank.full()) {
     SourceSite site;
```

Comment:
It would be nice if we had `Particle::to_source_site`, since this pattern is probably used elsewhere. Again, not necessary.

Reply:
Good suggestion! I'll look into doing this on a separate PR